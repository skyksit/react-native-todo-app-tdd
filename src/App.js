import React from 'react';
import {StyleSheet, Text, TextInput, View} from 'react-native';

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Todo TDD</Text>
      <TextInput placeholder="+ Add a Task" />
      <Text>Learn React Native</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
