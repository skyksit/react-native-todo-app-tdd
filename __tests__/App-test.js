import 'react-native';
import React from 'react';
import {render, screen} from '@testing-library/react-native';

import App from '../src/App';

let props;
let component;

function getAppComponent(temprops) {
  return <App {...temprops} />;
}

describe('App render', () => {
  beforeEach(() => {
    props = {};
    component = render(getAppComponent(props));
  });

  // it('snapshot test', () => {
  //   const snapshot = component.toJSON();
  //   expect(snapshot).toMatchSnapshot();
  // });

  it('should render title Todo TDD', () => {
    const title = screen.getByText('Todo TDD');
    expect(title).toBeTruthy();
  });

  it('should render AddTask', () => {
    const element = screen.getByPlaceholderText('+ Add a Task');
    expect(element).toBeTruthy();
  });

  it('should render TaskList', () => {
    const element = screen.getByText('Learn React Native');
    expect(element).toBeTruthy();
  });
});
