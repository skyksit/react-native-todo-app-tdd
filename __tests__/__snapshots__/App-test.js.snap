// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render snapshot test 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 40,
        "fontWeight": "600",
        "margin": 30,
      }
    }
  >
    Todo TDD
  </Text>
  <TextInput
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="+ Add a Task"
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 20,
        "margin": 5,
        "padding": 8,
        "width": 735,
      }
    }
    value=""
  />
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginLeft": 8,
            "width": 735,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "color": "green",
              "margin": 10,
              "opacity": 1,
            }
          }
          testID="checkbox"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "green",
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "black",
              "flex": 1,
              "fontSize": 20,
            }
          }
        >
          Learn React Native
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "blue",
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="edit"
        >
          
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "margin": 10,
              "opacity": 1,
            }
          }
          testID="delete"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "red",
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginLeft": 8,
            "width": 735,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "color": "green",
              "margin": 10,
              "opacity": 1,
            }
          }
          testID="checkbox"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "green",
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "black",
              "flex": 1,
              "fontSize": 20,
            }
          }
        >
          Create a React Native app
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "blue",
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="edit"
        >
          
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "margin": 10,
              "opacity": 1,
            }
          }
          testID="delete"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "red",
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
